---
title: "P8139_GWAS"
author: "Ruixi Li"
date: "2024-03-26"
output: html_document
---

```{r library, include=FALSE}
library(genetics)
```

```{r}
# read in data
fms = read.csv("FMS_data.csv")

# Chi-squared test for association: Suppose we are interested in
# determining whether there is an association between any of the SNPs within
# the esr1 gene and an indicator of body mass index (BMI) > 25 at baseline
# of the FAMuSS study. We use the following code first to identify the names
# of all of the esr1 SNPs:
attach(fms)
NamesEsr1Snps = names(fms)[substr(names(fms),1,4)=="esr1"]
NamesEsr1Snps
# The genotype matrix can now be defined by selecting the columns of fms that
# correspond to the esr1 SNP names:
fmsEsr1 = fms[,is.element(names(fms),NamesEsr1Snps)]

# We define our trait to be an indicator for whether BMI is > 25 at baseline:
Trait = as.numeric(pre.BMI>25)

# We write a function to record the p-values from applying the χ2-test to
# the 2 × 3 contingency tables corresponding to each SNP and this trait:
newFunction = function(Geno){
    ObsTab = table(Trait,Geno)
    return(chisq.test(ObsTab)$p.value)
}

# Apply this function to the columns of fmsEsr1:
pvalues = apply(fmsEsr1,2,newFunction)

# adjust for multiple testing
p.adj = p.adjust(pvalues, method="BH")

# Examine population substructure using all 24 SNPs within the akt1 gene.
NamesAkt1Snps = names(fms)[substr(names(fms),1,4)=="akt1"]
NamesAkt1Snps

# Convert the genotype data from factor variables to numeric variables using data.matrix()
# Note that we additionally assign the missing data a number

FMSgeno = fms[,is.element(names(fms),NamesAkt1Snps)]
FMSgenoNum = data.matrix(FMSgeno)
FMSgenoNum[is.na(FMSgenoNum)] = 4

PC.FMS = prcomp(FMSgenoNum)
plot(PC.FMS$"x"[,1],PC.FMS$"x"[,2],xlab="PC1",ylab="PC2")
```

